include/glfw/window/window.h

```diff
@@ -0,0 +1,163 @@
+/*
+ * -- GLFW thread safety key points --
+ *
+ * Initialization, termination, event processing and the creation and destruction of windows, cursors
+ * and OpenGL and OpenGL ES contexts are all restricted to the main thread.
+ *
+ * Because event processing must be performed on the main thread, all callbacks except for the error callback
+ * will only be called on that thread.
+ *
+ * All Vulkan related functions may be called from any thread.
+ *
+ */
+
+#pragma once
+
+#include <functional>
+#include <iostream>
+#include <memory>
+#include <span>
+#include <spdlog/spdlog.h>
+#include <string>
+
+#include "common/glfw_include.h"
+#include "error.h"
+
+namespace wnd
+{
+
+class Window;
+class WindowManager;
+
+struct WindowConfig;
+
+using WindowType = GLFWwindow;
+using FramebufferSizeType = std::pair<int, int>;
+
+using WindowResizeCallbackSignature = void(
+    Window* window,
+    int width,
+    int height //
+);
+
+struct WindowAPIversion
+{
+    int major;
+    int minor;
+
+    auto operator<=>( const WindowAPIversion& ) const = default;
+}; // WindowAPIversion
+
+struct WindowConfig
+{
+    const int width = k_default_width;
+    const int height = k_default_height;
+    const char* title = k_default_title;
+    const bool fullscreen = k_default_fullscreen;
+
+    std::function<WindowResizeCallbackSignature> resize_callback = defaultResizeCallback;
+
+    static constexpr int k_default_width = 640;
+    static constexpr int k_default_height = 480;
+    static constexpr char k_default_title[] = "MinCraft";
```

Comment:
This can be a ```std::string_view```. See https://en.cppreference.com/w/cpp/string/basic_string_view/basic_string_view
As you can see, this class is constexpr capable

Reply:
Simplicity and readability. So I prefer in that situation `char[]`.